examples/example-xds/src/main/java/io/grpc/examples/helloworldxds/XdsHelloWorldServer.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworldxds;
+
+import io.grpc.InsecureServerCredentials;
+import io.grpc.Server;
+import io.grpc.ServerCredentials;
+import io.grpc.health.v1.HealthCheckResponse.ServingStatus;
+import io.grpc.protobuf.services.ProtoReflectionService;
+import io.grpc.services.HealthStatusManager;
+import io.grpc.xds.XdsServerBuilder;
+import io.grpc.xds.XdsServerCredentials;
+import java.util.Arrays;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An xDS-managed Server for the {@code Greeter} service.
+ */
+public class XdsHelloWorldServer {
+  public static void main(String[] args) throws IOException, InterruptedException {
+    int port = 50051;
+    String hostname = null;
+    ServerCredentials credentials = InsecureServerCredentials.create();
+    if (args.length >= 1 && "--secure".equals(args[0])) {
+      // The xDS credentials use the security configured by the xDS server when available. When xDS
+      // is not used or when xDS does not provide security configuration, the xDS credentials fall
+      // back to other credentials (in this case, InsecureServerCredentials).
+      credentials = XdsServerCredentials.create(InsecureServerCredentials.create());
+      args = Arrays.copyOfRange(args, 1, args.length);
+    }
+    if (args.length >= 1) {
+      try {
+        port = Integer.parseInt(args[0]);
+      } catch (NumberFormatException ex) {
+        System.err.println("Usage: [--secure] [PORT [HOSTNAME]]");
+        System.err.println("");
+        System.err.println("  --secure  Use credentials provided by xDS. Defaults to insecure");
+        System.err.println("  PORT      The listen port. Defaults to " + port);
+        System.err.println("  HOSTNAME  The name clients will see in greet responses. ");
+        System.err.println("            Defaults to the machine's hostname");
+        System.exit(1);
+      }
+    }
+    if (args.length >= 2) {
+      hostname = args[1];
+    }
+    // Since the main server may be using TLS, we start a second server just for plaintext health
+    // checks
+    int healthPort = port + 1;
+    final HealthStatusManager health = new HealthStatusManager();
+    final Server server = XdsServerBuilder.forPort(port, credentials)
+        .addService(new HostnameGreeter(hostname))
+        .addService(ProtoReflectionService.newInstance()) // convenient for command line tools
+        .addService(health.getHealthService()) // allow management servers to monitor health
+        .build()
+        .start();
+    final Server healthServer =
+        XdsServerBuilder.forPort(healthPort, InsecureServerCredentials.create())
```

Comment:
Do we have to use `XdsServerBuilder` here? Without xDS configuration for this server/port the server won't start.

Reply:
I think we'd expect the management server to provide (default) configuration, for things like (IP?) access control and rate limiting, or at least logging and stats. If we are concerned about the basic non-xdscred case, we could only spin up the second server when xdscreds are used. But that doesn't seem right. We may need to discuss about this more.

Reply:
Hmmm, I didn't realize the management server will provide default configuration when the service (i.e. the port etc) is not configured (even if security is not required). Let me find out about the one management server (TD) we typically test with.

Reply:
Okay so TD apparently will return a default listener so that's good.